docs/DeveloperGuide.md

```diff
@@ -985,11 +1059,81 @@ Success action: When a person is successfully assigned:
        Expected (No Alice Tan): No person to unassign. Error details is shown in the status message, as the NAME does not belong to anyone in the address book.
 
 
-
 ### Saving data
 
 1. Dealing with missing/corrupted data files
 
-   1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
-
-1. _{ more test cases …​ }_
+    1. Prerequisites: Have the app running with some sample data saved
+
+    2. Test case: Delete addressbook.json
+        * Steps:
+            1. Navigate to data folder (default: `[App Home]/data/`)
+            2. Delete `addressbook.json`
+            3. Restart the application
+        * Expected:
+            - App creates new `addressbook.json` with empty data
+            - Status bar shows "Starting with empty address book"
+            - List shows no persons or weddings
+
+    3. Test case: Corrupt addressbook.json with invalid JSON
+        * Steps:
+            1. Navigate to data folder
+            2. Open `addressbook.json`
+            3. Add random text or remove brackets to make it invalid JSON
+            4. Save file and restart application
+        * Expected:
+            - Creates new `addressbook.json` with empty data
+            - Status bar shows "Corrupted data found. Starting with empty address book"
+            - List shows no persons or weddings
+
+    4. Test case: Corrupt addressbook.json with valid JSON but invalid data
+        * Steps:
+            1. Navigate to data folder
+            2. Open `addressbook.json`
+            3. Modify data while keeping valid JSON structure (e.g., remove required fields, change data types)
+            4. Save file and restart application
+        * Expected:
+            - Creates new `addressbook.json` with empty data
+            - Status bar shows "Corrupted data found. Starting with empty address book"
+            - List shows no persons or weddings
+
+2. Testing data persistence
+
+    1. Test case: Normal shutdown
+        * Steps:
+            1. Modify data (add/edit/delete person or wedding)
+            2. Exit application through UI
+            3. Restart application
+        * Expected: All modifications are preserved
+
+    2. Test case: Forced shutdown
+        * Steps:
+            1. Modify data
+            2. Force close application (Task Manager/Force Quit)
+            3. Restart application
+        * Expected:
+            - Last saved state is recovered
```

Comment:
Maybe can specify last saved state is last valid command executed before force quit